### PR TITLE
test: more sequencer retry tests

### PR DESCRIPTION
**Pull request by CHr15F0x**
Looking at those retry tests I realized that timeouts weren't tested at all so I added two more test cases and reworked the server factories btw.

### CODE REVIEW

**Review by Mirko-von-Leipzig — APPROVED** [review]

File: crates/pathfinder/src/sequencer.rs [hunk 1]
```diff
@@ -1038,11 +1060,78 @@ mod tests {
...
+            .unwrap_err();
             assert_matches!(
                 error,
                 SequencerError::TransportError(te) => assert_eq!(te.status(), Some(StatusCode::SERVICE_UNAVAILABLE))
             );
         }
+
+        #[tokio::test]
+        #[traced_test]
+        async fn client_timeout() {
```
Comment (kkovaacs) [inline crates/pathfinder/src/sequencer.rs:1080]: Aren't these two test basically the same?

I mean the only difference seems to be is that for `client_timeout()` the request timeout is set via the `ClientBuilder` and for `request_timeout()` via `RequestBuilder`. The result should be effectively the same, so I don't quite see the point in having two separate test cases.

Comment (CHr15F0x) [reply crates/pathfinder/src/sequencer.rs:1080]: Indeed, it's the server that determines the type of timeout here, regardless of whichever client api is used. I'll make it a single test case :+1:

File: crates/pathfinder/src/sequencer.rs [hunk 2]
```diff
@@ -1119,6 +1090,7 @@ mod tests {
                 let mut cnt = timeout_counter.lock().await;
                 *cnt += 1;
 
+                // Btw this is the same as using Client::builder().timeout()
```
Comment (Mirko-von-Leipzig) [inline crates/pathfinder/src/sequencer.rs:1093]:
```suggestion
                // This is the same as using Client::builder().timeout()
```

**Comment by CHr15F0x** [comment]
Test failure is a result of starknet updated to cairo-0.8, which will be sorted out in the next PR.

**Pull request merged**